framework/PostProcessors/Metric.py

```diff
@@ -0,0 +1,289 @@
+      outputExtension = output.getExt().lower()
+      if outputExtension not in availExtens:
+        self.raiseAMessage('Metric postprocessor did not recognize extension ".', str(outputExtension), '". The output will be dumped to a text file')
+      output.setPath(self.__workingDir)
+      self.raiseADebug('Write Metric prostprocessor output in file with name: ', output.getAbsFile())
+      output.open('w')
+      if outputExtension == 'xml':
+        self._writeXML(output, outputDict)
+      else:
+        separator = ' ' if outputExtension != 'csv' else ','
+        self._writeText(output, outputDict, separator)
+    else:
+      self.raiseAnError(IOError, 'Output type ', str(output.type), ' can not be used for postprocessor', self.name)
+
+  def _writeXML(self,output,outputDictionary):
+    """
+      Defines the method for writing the post-processor to a .csv file
+      @ In, output, File object, file to write to
+      @ In, outputDictionary, dict, dictionary stores importance ranking outputs
+      @ Out, None
+    """
+    if output.isOpen():
+      output.close()
+    if self.dynamic:
+      outputInstance = Files.returnInstance('DynamicXMLOutput', self)
+    else:
+      outputInstance = Files.returnInstance('StaticXMLOutput', self)
+    outputInstance.initialize(output.getFilename(), self.messageHandler, path=output.getPath())
+    outputInstance.newTree('MetricPostProcessor', pivotParam=self.pivotParameter)
+    outputResults = [outputDictionary] if not self.dynamic else outputDictionary.values()
+    for ts, outputDict in enumerate(outputResults):
+      pivotVal = outputDictionary.keys()[ts]
+      for nodeName, nodeValues in outputDict.items():
+        for metricName, value in nodeValues.items():
+          if type(value) == float:
+            outputInstance.addScalar(nodeName, metricName, value, pivotVal=pivotVal)
+          elif type(value) in [list, np.ndarray]:
+            if len(list(value)) == 1:
+              outputInstance.addScalar(nodeName, metricName, value[0], pivotVal=pivotVal)
+            else:
+              self.raiseAnError(IOError, "Multiple values are returned from metric '", metricName, "', this is currently not allowed")
+          elif type(value) == dict:
+            ## FIXME: The following are used to accept timedependent data, and should be checked later.
+            outputInstance.addVector(nodeName, metricName, value, pivotVal=pivotVal)
+          else:
+            self.raiseAnError(IOError, "Unrecognized type of input value '", type(value), "'")
+    outputInstance.writeFile()
+
+  def _writeText(self,output,outputDictionary, separator=' '):
+    """
+      Defines the method for writing the post-processor to a .csv file
+      @ In, output, File object, file to write to
+      @ In, outputDictionary, dict, dictionary stores importance ranking outputs
+      @ In, separator, string, optional, separator string
+      @ Out, None
+    """
+    ##TODO: Add this method if needed
+    pass
```

Comment:
If this is not implemented, it would be better to throw an error than let it pass silently. (Otherwise you mystify the user as to why their text output is not written)